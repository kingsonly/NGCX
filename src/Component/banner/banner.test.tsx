/**
 * @jest-environment jsdom
 */

import "@testing-library/jest-dom";

import renderer from "react-test-renderer";
import {
  render,
  screen,
  fireEvent,
  queryByLabelText,
  queryByTestId,
} from "@testing-library/react";

import React from "react";

import { FiMenu, FiX } from "react-icons/fi";
import { SubscriptionInput } from ".";
import userEvent from "@testing-library/user-event";

describe("Test Banner Component", () => {
  test("Renders Subscription Input", () => {
    const component = render(<SubscriptionInput></SubscriptionInput>);
    expect(component).toMatchSnapshot();
    const element = screen.getByTestId('email')
    expect(element).toBeInTheDocument()
  });


//   test('pass valid email to Subscription Input', () => {
//     render(<SubscriptionInput />);
 
//     const inputEl = screen.getByTestId("email");
//     userEvent.type(inputEl, "test@mail.com");
 
//     expect(screen.getByTestId("email-input")).toHaveValue("test@mail.com");
//   });
});
