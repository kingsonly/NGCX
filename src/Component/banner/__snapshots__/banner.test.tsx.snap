// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Banner Component Renders Subscription Input 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex bg-white items-center rounded-full w-2/3 pl-4"
      >
        <svg
          color="#7E7E7E"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          style="color: rgb(126, 126, 126);"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="22"
            x2="11"
            y1="2"
            y2="13"
          />
          <polygon
            points="22 2 15 22 11 13 2 9 22 2"
          />
        </svg>
        <input
          class="placeholder:text-ncgx-text-lightgray placeholder:text-xs text-xs outline-none w-full ml-4"
          data-testid="email"
          placeholder="Your email address"
        />
        <button
          class="text-white font-semibold text-sm bg-ncgx-green rounded-full px-5 py-2 outline-none "
        >
          Subscribe
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex bg-white items-center rounded-full w-2/3 pl-4"
    >
      <svg
        color="#7E7E7E"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        style="color: rgb(126, 126, 126);"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="22"
          x2="11"
          y1="2"
          y2="13"
        />
        <polygon
          points="22 2 15 22 11 13 2 9 22 2"
        />
      </svg>
      <input
        class="placeholder:text-ncgx-text-lightgray placeholder:text-xs text-xs outline-none w-full ml-4"
        data-testid="email"
        placeholder="Your email address"
      />
      <button
        class="text-white font-semibold text-sm bg-ncgx-green rounded-full px-5 py-2 outline-none "
      >
        Subscribe
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
